src/types.jl

```diff
@@ -528,28 +535,12 @@ end
 
 unflatten(t) = t
 
-"""
-    similarterm(t, f, args, symtype; metadata=nothing)
-
-Create a term that is similar in type to `t`. Extending this function allows packages
-using their own expression types with SymbolicUtils to define how new terms should
-be created. Note that `similarterm` may return an object that has a
-different type than `t`, because `f` also influences the result.
-
-## Arguments
+function TermInterface.maketerm(::Type{<:BasicSymbolic}, head, args, type, metadata)
+    basicsymbolic(first(args), args[2:end], type, metadata)
+end
```

Comment:
Please let's use the new one directly

Reply:
```
function TermInterface.maketerm(::Type{<:BasicSymbolic}, op, args, type=nothing, metadata=nothing)
    basicsymbolic(op, args, type, metadata)
end
```

Reply:
xref https://github.com/JuliaSymbolics/TermInterface.jl/issues/33